tests/common/platform/transceiver_utils.py

```diff
@@ -133,13 +133,270 @@ def check_transceiver_dom_sensor_details(dut, asic_index, interfaces, xcvr_skip_
+    previous_key = ""
+    top_key = ""
+    for line in sfp_eeprom_info[start:end+1]:
+        res1 = re.match(pattern_top_layer_key_value, line)
+        if res1:
+            top_key = res1.groupdict()["key"].strip()
+            one_sfp_eeprom_info_dict[top_key] = res1.groupdict()[
+                "value"].strip()
+            continue
+        res2 = re.match(pattern_second_layer_key_value, line)
+        if res2:
+            if second_layer_dict and previous_key:
+                one_sfp_eeprom_info_dict[previous_key] = second_layer_dict
+                second_layer_dict = {}
+            one_sfp_eeprom_info_dict[res2.groupdict()["key"].strip()] = res2.groupdict()[
+                "value"].strip()
+            previous_key = res2.groupdict()["key"].strip()
+        else:
+            res3 = re.match(pattern_third_layer_key_value, line)
+            if res3:
+                second_layer_dict[res3.groupdict()["key"].strip()] = res3.groupdict()[
+                    "value"].strip()
+    if second_layer_dict and previous_key:
+        one_sfp_eeprom_info_dict[previous_key] = second_layer_dict
+
+    return top_key, one_sfp_eeprom_info_dict
+
+
+def parse_sfp_eeprom_infos(eeprom_infos):
+    """
+    This method is to pares sfp eeprom infos, and return sfp_eeprom_info_dict
+    """
+    map_port_to_start_and_end_line_nubmer, sfp_eeprom_list = get_map_port_to_start_and_end_line_number_for_sfp_eeporm(
+        eeprom_infos)
+    sfp_eeprom_info_dict = {}
+    for port_name, line_number_section in map_port_to_start_and_end_line_nubmer.items():
+        _, eeprom_info = parse_one_sfp_eeprom_info(sfp_eeprom_list, line_number_section[0], line_number_section[1])
+        sfp_eeprom_info_dict[port_name] = eeprom_info
+    return sfp_eeprom_info_dict
+
+
+def is_passive_cable(sfp_eeprom_info):
+    # The implementation of the function refers to the function of is_xcvr_optical
+    if "Specification compliance" in sfp_eeprom_info:
+        spec_compliance = sfp_eeprom_info["Specification compliance"]
+
+        if "passive" in spec_compliance or "Passive" in spec_compliance:
+            # for QSFP-DD  OSFP-8X, and QSFP+C, it has the key "Specification compliance"
+            return True
+        elif isinstance(spec_compliance, dict):
+            if "SFP+CableTechnology" in spec_compliance and "Passive" in spec_compliance.get("SFP+CableTechnology", ""):
+                # For SFP/SFP+/SFP28, it has the key  "SFP+CableTechnolog"
+                return True
+            else:
+                if "10/40G Ethernet Compliance Code" in spec_compliance and\
+                        "CR" in spec_compliance.get("10/40G Ethernet Compliance Code", " "):
+                    return True
+                if "Extended Specification Compliance" in spec_compliance and\
+                        "CR" in spec_compliance.get("Extended Specification Compliance", " "):
+                    return True
+    return False
+
+
+def get_passive_cable_port_list(dut):
```

Comment:
@JibinBao Where is this function invoked?

Reply:
Remove it